File: aux_api/lib/aux_api/qentry.ex
defmodule AuxApi.Qentry do
  use Ecto.Schema
  import Ecto.Changeset

  schema "qentries" do
    field :song_id, :string

    field :played, :boolean, default: false
    field :next_qentry_id, :integer
    field :prev_qentry_id, :integer

    belongs_to :member, AuxApi.Member
    belongs_to :session, AuxApi.Session

    timestamps()
  end

  @doc false
  def changeset(qentry, attrs) do
    qentry
    |> cast(attrs, [:is_host, :song_id, :has_played])
    |> validate_required([:is_host, :song_id, :has_played])
  end
end

File: aux_ui/lib/aux_lib/qentry.dart
class Qentry {
  int id;
  String trackId;

  Qentry.fromList(List<dynamic> schema) {
    this.id = schema[0];
    this.trackId = schema[1];
  }

  Qentry(int id, String trackId) {
    this.id = id;
    this.trackId = trackId;
  }
}
